src/server/scripts/Kalimdor/zone_durotar.cpp

```diff
@@ -177,9 +178,535 @@ struct npc_mithaka : ScriptedAI
+    }
+
+    bool OnGossipSelect(Player* player, uint32 /*menuId*/, uint32 /*gossipListId*/) override
+    {
+        player->KilledMonsterCredit(NPC_DARKSPEAR_JAILOR);
+        Talk(SAY_GET_IN_THE_PIT, player);
+        CloseGossipMenuFor(player);
+        _eventInProgress = true;
+        _events.ScheduleEvent(EVENT_PLAYER_ACCEPT_CHALLENGE, 2s);
+        return true;
+    }
+
+    bool OnGossipHello(Player* player) override
+    {
+        if (_eventInProgress)
+        {
+            //AddGossipItemFor(player, GOSSIP_EVENT_IN_PROGRESS_ID, 0, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF);
+            SendGossipMenuFor(player, TEXT_GOSSIP_EVENT_IN_PROGRESS, me->GetGUID());
+            return true;
+        }
+        return false;
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_PLAYER_ACCEPT_CHALLENGE:
+                    me->GetMotionMaster()->MovePath(_pathCage, false);
+                    break;
+                case EVENT_WALK_BACK_TO_HOME:
+                    if (Creature* scout = me->FindNearestCreature(NPC_CAPTIVE_SPITESCALE_SCOUT, 5.0f, true))
+                        scout->AI()->SetData(1, 1);
+                    me->GetMotionMaster()->MovePath(_pathHome, false);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+private:
+    EventMap _events;
+    bool _eventInProgress;
+    uint32 _pathCage;
+    uint32 _pathHome;
+    float _facing;
+};
+
+enum CaptiveSpitescaleScout
+{
+    EVENT_UPDATE_JAILOR_GOSSIP    = 1,
+    EVENT_TALK_TO_PLAYER          = 2,
+    EVENT_DESPAWN_OUT_OF_COMBAT   = 3,
+    EVENT_CAST_FROSTBOLT          = 4,
+
+    SAY_SSEND_YOU_TO_YOUR_DEATH   = 0,
+
+    SPELL_FROST_SHOCK             = 15089
+};
+
+struct npc_captive_spitescale_scout : public ScriptedAI
+{
+    npc_captive_spitescale_scout(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        _events.ScheduleEvent(EVENT_UPDATE_JAILOR_GOSSIP, 2s);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+        me->DespawnOrUnsummon(10s, 7s);
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        me->DespawnOrUnsummon(0s, 2s);
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        _events.CancelEvent(EVENT_DESPAWN_OUT_OF_COMBAT);
+        _events.ScheduleEvent(EVENT_CAST_FROSTBOLT, 2s, 4s);
+    }
+
+    void SetData(uint32 /*type*/, uint32 data) override
+    {
+        if (data == 1)
+        {
+            me->SetWalk(true);
+
+            if (me->HasStringId("captive_spitescale_scout_one"))
+                me->GetMotionMaster()->MovePoint(0, -1142.49f, -5415.59f, 10.597655f);
```

Comment:
move position to constant variable,
Define a name for 0 in enum (POINT_WHATEVER_DESCRIBES_IT) if you are planning to use it in MovementInform

Reply:
Done